### PR TITLE
New: added unix-style formatter (fixes #2991)

### LINKED ISSUE #2991: Add unix-style formatter
A formatter which uses more traditional ~~grep~~ unix-style output such as this:

directory/file.js:6:2: Strings must use singlequote. [Error/quotes]

should in my opinion be included in the base eslint distribution - all major linters already support this format (pyflakes, rubocop, perlcritic, not to mention jshint) because it's a standard which can be understood by tools and editors such as Emacs and it's also very easy to parse when using grep/sed/awk.

### PR DESCRIPTION
**Pull request by oshi-shinobu**
I wrote the formatter described in #2991 and the tests for it, and released it under the same free license as eslint. I hope you will consider distributing it with eslint, as it follows a standardized format adopted by most linters and editors.

### CODE REVIEW

**Comment by lo1tuma** [comment]
Thanks for the PR. Please take a moment and read our [contributing guidelines](http://eslint.org/docs/developer-guide/contributing). We require to have an open issue before submitting a pull request.

**Comment by oshi-shinobu** [comment]
I apologize, everything should be good now.

Only one thing - I'm fine with signing the CLA but I don't want to do it through Google, can we please do it via email? Thank you.

File: lib/formatters/grep.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/**
+ * @fileoverview grep-style formatter.
+ * @author Nicholas C. Zakas
```
Comment (nzakas) [inline lib/formatters/grep.js]: Put your name here.

File: lib/formatters/grep.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ */
+"use strict";
+
+//-----------------------------------------------------------------------------
+// Helper Functions
+//-----------------------------------------------------------------------------
+
+function getMessageType(message) {
```
Comment (nzakas) [inline lib/formatters/grep.js]: Please add JSDoc comment for this function

File: tests/lib/formatters/grep.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
+/**
+ * @fileoverview Tests for grep-style formatter.
+ * @author Nicholas C. Zakas
```
Comment (nzakas) [inline tests/lib/formatters/grep.js]: Put your name here.

**Comment by nzakas** [comment]
Thanks for the PR. Some feedback:

1. I think this should be called "unix" instead of "grep" to match JSHint
2. Please add "New:" to the front of your commit message
3. Please update the docs: https://github.com/eslint/eslint/blob/master/docs/user-guide/command-line-interface.md#-f---format
4. Please sign our CLA: http://eslint.org/cla

File: lib/formatters/grep.js (hunk 2 above)
Comment (oshi-shinobu) [inline lib/formatters/grep.js]: The function is actually part of lib/formatters/compact.js which you wrote, could you please provide the JSDoc for it and/or should I write it?

File: lib/formatters/grep.js (hunk 1 above)
Comment (oshi-shinobu) [inline lib/formatters/grep.js]: Is it necessary? I am already telling you my real name in the CLA so you've got the copyright covered.

**Comment by oshi-shinobu** [comment]
You are welcome!

I did 1, 2 and 3 I can sign a CLA but I really am not comfortable with using Google services, can I please do it via email? You can write me at the no-reply address in the profile and I will get back to you with all the details.

**Comment by nzakas** [comment]
@oshi-shinobu while you can send your information via email, I will still need to add it to the Google spreadsheet we use for tracking this information. Are you okay with that?

**Comment by oshi-shinobu** [comment]
What people do with their data is their own business, even if such data involves me, so please go ahead.

File: tests/lib/formatters/unix.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
+/**
+ * @fileoverview Tests for unix-style formatter.
+ * @author Nicholas C. Zakas
```
Comment (IanVS) [inline tests/lib/formatters/unix.js]: @oshi-shinobu you should put your name and copyright here, as you are the author.

File: lib/formatters/unix.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
+/**
+ * @fileoverview unix-style formatter.
+ * @author Nicholas C. Zakas
```
Comment (IanVS) [inline lib/formatters/unix.js]: @oshi-shinobu you should put your name and copyright here.

File: lib/formatters/unix.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Helper Functions
+//------------------------------------------------------------------------------
+
+function getMessageType(message) {
```
Comment (IanVS) [inline lib/formatters/unix.js]: Can you please add JSDoc here?  See https://github.com/eslint/eslint/blob/master/lib/formatters/tap.js#L13-L17 for an example.

**Comment by nzakas** [comment]
@oshi-shinobu I tried the no-reply email and it bounced. Can you send your info to info (at) eslint (dot) org?

**Comment by nzakas** [comment]
@oshi-shinobu just checking in again, can you let us know if you're going to finish this up?

**Comment by oshi-shinobu** [comment]
I sent you a mail around a week ago but it bounced back with a "Undelivered mail returned to sender" some days ago. Are you sure info (at) eslint (dot) org is a valid address? Else, could you send me a mail at the address specified in my profile?

**Comment by nzakas** [comment]
Just sent you an email. In the meantime, can you rebase?

**Comment by oshi-shinobu** [comment]
Done.

**Pull request merged**